lib/tdf3/src/crypto/decrypt-worker.ts

```diff
@@ -0,0 +1,75 @@
+import { TdfDecryptError } from '../../../src/errors.js';
+
+const maxWorkers = navigator?.hardwareConcurrency || 4; // save fallback number
+
+interface DecryptData {
+  key: CryptoKey;
+  encryptedPayload: ArrayBuffer;
+  algo: AesCbcParams | AesGcmParams;
+}
+
+const workerScript = `
+  self.onmessage = async (event) => {
+    const { key, encryptedPayload, algo } = event.data;
+
+    try {
+      const decryptedData = await crypto.subtle.decrypt(algo, key, encryptedPayload);
+      self.postMessage({ success: true, data: decryptedData });
+    } catch (error) {
+      self.postMessage({ success: false, error: error.message });
+    }
+  };
+`;
+
+const workerBlob = new Blob([workerScript], { type: 'application/javascript' });
+const workerUrl = URL.createObjectURL(workerBlob);
+const workersArray: Worker[] = Array.from({ length: maxWorkers }, () => new Worker(workerUrl));
```

Comment:
Ideally we'd be doing this at build time, not runtime. Do we have any methods for doing this? I can't find a good way to do this without doing a two-pass build.

Reply:
This code gets executate in the instance when we import entry point of sdk to any frontend app. Because this file eventually gets imported from entry point further down the nodes. And imports are synchronous. ANd code above too.

Reply:
Oh yes i tried it and i had constant problems on a app side. Ive tuned it on Virtru-sdk side but after webpack on secure share it had constant errors. Spend a wile on that. 

Thats why i decided to go with this aproach. Its makes sdk independent of bundlers issues and doesnt slow down performance, since its a singleton instance that initializes workers as soon as sdk imported it. So when youll start download stuff its already be there.